### DISCUSSION TITLE
Error opening a TLS connection on SQL Server 2017

### DISCUSSION THREAD

**Discussion by thesobercoder** [prisma/prisma#5817] [Q&A] [answered]
I'm getting this error when running `npx prisma introspect`

```
Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider. (os error -2146762487))
```

I'm on my work laptop and am aware that there are some self-signed certificates installed. Is there a workaround for this?

My environment - 

Windows 10
NodeJS 15.10.0
Prisma 2.17.0

**Comment by thesobercoder** [comment, marked as answer]
Nevermind. I figured it out via this connection string - 

`sqlserver://127.0.0.1:1433;database=database;user=user;password=password;encrypt=DANGER_PLAINTEXT`

**Reply by pimeys** [reply]
Hey. Please don't use the `DANGER_PLAINTEXT` option, that will mean even your database credentials are sent plaintext over the wire. It's fine if it's your own machine, but if you have any network in between, you should maybe consider changing your database password right now.

There are a few options you can use, so you can continue using TLS but don't need a valid certificate:

- `encrypt` can be either `true` or `false`. The former will encrypt all traffic, the latter only the login details. Nowadays we have fast cpus for TLS, so I'd recommend setting this to `true`.
- `trustServerCertificate` is a setting that defines do we need a valid cert or not. On Azure SQL and if you have a public domain with a signed cert to the database, this should be `false`. Otherwise, if self-signing, set this to `true` and you should be able to connect.

**Reply by mustafa-ahmed-dev** [reply]
Thank you so much, you saved me a lot of search

**Reply by solmon** [reply]
Thank you so much ..

**Reply by YourfriendRND** [reply]
Thank you so much !

**Reply by Enauro-Web** [reply]
this fix also worked for MSSQL Server 2022. Thanks a lot!

**Comment by jagdishpadeliya** [comment]
Connection String : sqlserver://Server-Address;database=Database_Name;user=Username;password=Password;encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
